### PR TITLE
calamares: fix python modules

### PR DESCRIPTION
**Pull request by Gaelan**
###### Motivation for this change

Calamares wants a boost with python support. It builds without it, but omits any python modules, which are a lot of the interesting ones. This PR also incorporates #114016, which is necessary to build calamares on master.

(As a side note, it's a shame that kpmcore runs on DBus now; as of a few weeks ago, it was possible to run Calamares without any reliance on the global system, which would have made it possible to use any system with Nix on it as a NixOS installer. Ah well.)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115344` at c7ae8524 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.kpmcore</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/6e85e194938c203f25859b7b91e8dc65">calamares</a></li></li>
    <li><a href="https://gist.github.com/56d62d126eb3c0e9f93e5fee0a25af32">libsForQt5.kpmcore</a></li> (libsForQt515.kpmcore ,plasma5Packages.kpmcore)</li>
    <li><a href="https://gist.github.com/487c2441bdd64b5b0db81c23146248e0">libsForQt514.kpmcore</a></li></li>
    <li><a href="https://gist.github.com/970a1fc3caf0fead2da6e699ec825f72">partition-manager</a></li></li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115344` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>libsForQt512.kpmcore</li>
</ul>
</details>
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>calamares</li>
<li>libsForQt5.kpmcore (libsForQt515.kpmcore ,plasma5Packages.kpmcore)</li>
<li>libsForQt514.kpmcore</li>
<li>partition-manager</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
calamares:
<details>


warning: build-tools-in-build-inputs
util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/misc/calamares/default.nix:18:3:

```
   |
18 |   buildInputs = [
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
not sure about the module.

**Comment by Gaelan** [comment]
The module is from https://github.com/NixOS/nixpkgs/pull/114016; that PR should probably be merged separately before this one

File: nixos/modules/programs/partition-manager.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ config, lib, pkgs, ... }:
```
Comment (FRidh) [inline nixos/modules/programs/partition-manager.nix:1]: What value does this service add over just adding the package to your closure?

Comment (Gaelan) [reply nixos/modules/programs/partition-manager.nix:1]: That's from the PR https://github.com/NixOS/nixpkgs/pull/114016, which is required to make calamares build and is therefore included in this PR. The issue is that part of kpmcore is now a daemon called through DBus, so it needs a module to be register it on the bus. (grrr)

Comment (FRidh) [reply nixos/modules/programs/partition-manager.nix:1]: Right, I see the `services.dbus.packages`, and I see it has become a pattern to create modules for this as well as `systemd.packages`.

We don't have it yet, but we should tag packages as providing dbus and/or systemd files instead.

**Comment by Artturin** [comment]
Added python support to boost #136929

**Pull request closed without merging**